src/jaxsim/integrators/common.py

```diff
@@ -413,10 +422,10 @@ def compute_ki() -> jax.Array:
             K = jax.tree_util.tree_map(op, K, ki)
 
             carry = K
-            return carry, None
+            return carry, aux_dict
 
         # Compute the state derivatives kᵢ.
-        K, _ = jax.lax.scan(
+        K, aux_dict = jax.lax.scan(
```

Comment:
Doing so, the `aux_dict` would contain a batched dictionary stacking the dictionaries returned at all stages. It would be ok to return it as a whole to the user, but there is no easy way to understand the right row e.g. corresponding to $t_f$ if not by inspecting the active integrator (particularly, the `row_index_of_solution` attribute).

For the rigid contacts, what type of aux variable do you need and at what specific time within the integration step? I'm asking since I'm thinking to return here only the `aux_dict` corresponding to $t_f$ instead of the one corresponding to all times specified in the `c` vector of the Butcher Tableau of the active integrator.

Reply:
Thanks @diegoferigo for the explanation! I updated this part of integrators in order to be able to pass states of the rigid contacts that do not need to be integrated into the `model.step` function. 

This was necessary at the beginning when I needed to have the bool flag indicating if new impacts have been created in the current simulation step. However, we eventually discarded this information and proceeded without it. 

I left the modifications there because I thought they could be useful in the future to contain states that don't need to be integrated, but if you think that this needs further iteration, we could also discard this change.

Reply:
Ok!

Reply:
Done in 808356bc8308f499e65272b7d7c2adcded37413a